<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="slam_graph_topic"     default="/vilens_slam/slam_poses"/>
  <arg name="pose_topic"           default="/state_estimator/pose_in_odom"/>
  <arg name="twist_topic"          default="/state_estimator/twist"/>
  <arg name="vilens_state_topic"   default=""/>
  
  <arg name="tf_reference_frames"  default="['base', 'map_vilens']"/>
  <arg name="tf_query_frames"      default="['LF_FOOT', 'RF_FOOT', 'LH_FOOT', 'RH_FOOT']"/>

  <!-- Optional topics -->
  <arg name="reference_twist_topic" default="/local_guidance_path_follower/twist"/>
  <arg name="operator_twist_topic"   default="/motion_reference/command_twist"/>

  <!-- Launch mission_analysis_node -->
  <node pkg="digiforest_analysis_ros" type="mission_analysis_node.py" name="digiforest_mission_analysis" output="screen" >
    <rosparam command="delete" param="tf_reference_frames"/>
    <rosparam param="tf_reference_frames" subst_value="True">$(arg tf_reference_frames)</rosparam>

    <rosparam command="delete" param="tf_query_frames"/>
    <rosparam param="tf_query_frames" subst_value="True">$(arg tf_query_frames)</rosparam>

    <param name="slam_graph_topic"     value="$(arg slam_graph_topic)" />

    <param name="pose_topic"           value="$(arg pose_topic)" />
    <param name="twist_topic"          value="$(arg twist_topic)" />
    <param name="vilens_state_topic"   value="$(arg vilens_state_topic)" />
    
    <param name="reference_twist_topic"  value="$(arg reference_twist_topic)"/>
    <param name="operator_twist_topic"   value="$(arg operator_twist_topic)" />
  </node>

</launch>